great_expectations/data_context/store/inline_store_backend.py

```diff
@@ -183,8 +184,14 @@ def _has_key(self, key: Tuple[str, ...]) -> bool:
         return resource_type in self._data_context.config
 
     def _save_changes(self) -> None:
-        # NOTE: <DataContextRefactor> This responsibility will be moved into DataContext Variables object
-        self._data_context._save_project_config()
+        context = self._data_context
+        config_filepath = os.path.join(context.root_directory, context.GE_YML)
+
+        try:
+            with open(config_filepath, "w") as outfile:
+                context.config.to_yaml(outfile)
+        except PermissionError as e:
+            logger.warning(f"Could not save project config to disk: {e}")
```

Comment:
Move writing YAML to disk to this class - keeping this as-is would lead to an infinite loop as `_save_project_config` references the `variables.save_changes` method

Reply:
Is it ok that this save fails? That is, should we reraise? If not, could you add a comment?

Reply:
this was implemented as a bit of a hotfix due to calls accidentally writing to disk in environments where it wasn't allowed. I'll add a comment!